### ISSUE TITLE
LSP warning upon name shadowing

### ISSUE THREAD

**Issue by kostmo** [swarm-game/swarm#1338]
Sometimes name shadowing is fine and intentional.  However, I did accidentally define two different top-level functions (using `def`) with the same name, which silently led to unexpected results.  It would be nice for the IDE to warn of this specific mode of name shadowing.

**Comment by byorgey** [comment]
Yes, that sounds like a nice warning to have.  Should be pretty easy, just modify the code here:

https://github.com/swarm-game/swarm/blob/3bbbd8826172983b77e906d19b9cd39081091d08/src/Swarm/Language/Typecheck.hs#L506-L519

to first look up the name in the context and generate a warning if it already exists.

Hmm, actually, I don't know if we're currently set up to be able to be able to generate warnings from within the typechecker.  That's definitely something I want to work on though --- modifying the type checker so that it can generate multiple errors + warnings, rather than just stopping as soon as it encounters an error.
